menpo/transform/thinplatesplines.py

```diff
@@ -20,18 +20,24 @@ class ThinPlateSplines(Alignment, Transform, Invertible):
         The target points to apply the tps to
     kernel : :class:`menpo.transform.rbf.RadialBasisFunction`, optional
         The kernel to apply.
+    min_singular_val : `float`, optional
+        If the target has points that are nearly coincident, the coefficients
+        matrix is rank deficient, and therefore not invertible. Therefore, we
+        only take the inverse on the full-rank matrix and drop any singular
+        values that are less than this value (close to zero).
 
     Raises
     ------
     ValueError
         TPS is only with on 2-dimensional data
     """
-    def __init__(self, source, target, kernel=None):
+    def __init__(self, source, target, kernel=None, min_singular_val=1e-4):
         Alignment.__init__(self, source, target)
         if self.n_dims != 2:
             raise ValueError('TPS can only be used on 2D data.')
         if kernel is None:
             kernel = R2LogR2RBF(source.points)
+        self._min_singular_val = min_singular_val
```

Comment:
@patricksnape I think this would be better non '_' - imagine users would want to be able to check this parameter after constructing warps in the future